modules/300-prometheus/images/alerts-receiver/src/alerts-receiver.go

```diff
@@ -140,18 +138,46 @@ func reconcileLoop(ctx context.Context) {
 }
 
 func reconcile() {
-	alertStore.m.RLock()
-	defer alertStore.m.RUnlock()
-	for _, v := range alertStore.alerts {
-		f := v.alert.Fingerprint
+	log.Info("starting reconcile")
 
-		if err := alertStore.updateEvent(f); err != nil {
+	crSet, err := listCRs()
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	// Add or update CRs
+	alertSet := make(map[string]struct{}, len(alertStore.alerts))
+	for f, v := range alertStore.alerts {
+		if v.Resolved() {
+			alertStore.removeAlert(f)
+			continue
+		}
+
+		alertSet[f.String()] = struct{}{}
+
+		// is alerts CR does not exist in cluster, insert CR
+		if _, ok := crSet[f.String()]; !ok {
+			err := alertStore.insertCR(f)
+			if err != nil {
+				log.Error(err)
+			}
```

Comment:
I guess, you can add `continue` word here. If you have just inserted an alert, you don't need to update it on the line 168

Reply:
no, because at line 168 we update status of cr